--- a/libstdc++-v3/src/c++23/std.cc.in	2026-04-30 23:34:46
+++ b/libstdc++-v3/src/c++23/std.cc.in	2026-04-30 23:42:28
@@ -1368,17 +1368,21 @@
 #endif
   using std::formatter;
   using std::make_format_args;
+#if _GLIBCXX_USE_WCHAR_T
   using std::make_wformat_args;
+#endif
 #if __cpp_lib_format >= 202603L // >= C++26
   using std::dynamic_format;
 #endif
   using std::vformat;
   using std::vformat_to;
   using std::visit_format_arg;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wformat_args;
   using std::wformat_context;
   using std::wformat_parse_context;
   using std::wformat_string;
+#endif
 #ifdef __cpp_lib_format_ranges
   using std::format_kind;
   using std::range_format;
@@ -1416,10 +1420,12 @@
   using std::ifstream;
   using std::ofstream;
   using std::swap;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wfilebuf;
   using std::wfstream;
   using std::wifstream;
   using std::wofstream;
+#endif
 }
 
 // <functional>
@@ -1620,7 +1626,9 @@
   using std::streamsize;
   using std::unitbuf;
   using std::uppercase;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wios;
+#endif
 }
 
 // <iosfwd>
@@ -1651,10 +1659,12 @@
   using std::cin;
   using std::clog;
   using std::cout;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wcerr;
   using std::wcin;
   using std::wclog;
   using std::wcout;
+#endif
 }
 
 // <istream>
@@ -1664,8 +1674,10 @@
   using std::basic_istream;
   using std::iostream;
   using std::istream;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wiostream;
   using std::wistream;
+#endif
   using std::ws;
   using std::operator>>;
 }
@@ -2487,7 +2499,9 @@
   using std::ends;
   using std::flush;
   using std::ostream;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wostream;
+#endif
   using std::operator<<;
 #ifdef __cpp_lib_syncbuf
   using std::emit_on_flush;
@@ -2916,9 +2930,11 @@
   using std::ssub_match;
   using std::sub_match;
   using std::swap;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wcsub_match;
   using std::wregex;
   using std::wssub_match;
+#endif
   using std::operator==;
   using std::operator<=>;
   using std::operator<<;
@@ -2934,21 +2950,25 @@
   using std::smatch;
   using std::sregex_iterator;
   using std::sregex_token_iterator;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wcmatch;
   using std::wcregex_iterator;
   using std::wcregex_token_iterator;
   using std::wsmatch;
   using std::wsregex_iterator;
   using std::wsregex_token_iterator;
+#endif
   namespace pmr
   {
 #if _GLIBCXX_USE_CXX11_ABI
     using std::pmr::cmatch;
     using std::pmr::match_results;
     using std::pmr::smatch;
+#if _GLIBCXX_USE_WCHAR_T
     using std::pmr::wcmatch;
     using std::pmr::wsmatch;
 #endif
+#endif
   }
 }
 
@@ -3021,16 +3041,24 @@
 {
   using std::basic_spanbuf;
   using std::spanbuf;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wspanbuf;
+#endif
   using std::basic_ispanstream;
   using std::ispanstream;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wispanstream;
+#endif
   using std::basic_ospanstream;
   using std::ospanstream;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wospanstream;
+#endif
   using std::basic_spanstream;
   using std::spanstream;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wspanstream;
+#endif
 }
 #endif
 
@@ -3046,10 +3074,12 @@
   using std::stringbuf;
   using std::stringstream;
   using std::swap;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wistringstream;
   using std::wostringstream;
   using std::wstringbuf;
   using std::wstringstream;
+#endif
 }
 
 // <stack>
@@ -3144,7 +3174,9 @@
 {
   using std::basic_streambuf;
   using std::streambuf;
+#if _GLIBCXX_USE_WCHAR_T
   using std::wstreambuf;
+#endif
 }
 
 // <string>
@@ -3171,7 +3203,9 @@
   using std::stoull;
   using std::string;
   using std::to_string;
+#if _GLIBCXX_USE_WCHAR_T
   using std::to_wstring;
+#endif
   using std::u16string;
   using std::u32string;
   using std::u8string;
